src/util/heap/gc_trigger.rs

```diff
@@ -467,6 +480,10 @@ impl MemBalancerTrigger {
             (live as f64 * 4096f64).sqrt()
         };
 
+        // Take pending allocations into consideration
+        extra_reserve += self.pending_pages.load(Ordering::SeqCst);
```

Comment:
I think it is clearer to simply create a new variable for pending pages,

```suggestion
        let pending_pages = self.pending_pages.load(Ordering::SeqCst);
```
... and then add it to `optimal_heap`.

```diff
-let optimal_heap = live + e as usize + extra_reserve;
+let optimal_heap = live + e as usize + extra_reserve + pending_pages;
```

And we don't need to make the `extra_reserve` parameter mutable.